File: Array | Dictionary | Set/concatenation_of_array/js/concatenation_of_array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var getConcatenation = function(nums) {
    return nums.concat(nums);
};

var getConcatenation_II = function(nums) {

    let len = nums.length;

    for ( let i = 0 ; i < len ; i++ ) {
        nums[i + len] = nums[i];
        // nums.push(nums[i]);
    }

    return nums;
};

File: Array | Dictionary | Set/concatenation_of_array/py/concatenation_of_array.py
from typing import List

class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        nums.extend(nums)
        return nums
    
class Solution_II:
    def getConcatenation(self, nums: List[int]) -> List[int]:

        length = len(nums)

        for i in range(length):

            nums.append(nums[i])

        return nums
